seo_keyword_research.py

```diff
@@ -0,0 +1,120 @@
+        self.country = country
+        self.domain = domain
+
+
+    def get_auto_complete(self) -> list:
+        params = {
+            'api_key': self.api_key,            # https://serpapi.com/manage-api-key
+            'engine': 'google_autocomplete',    # search engine
+            'q': self.query,                    # search query
+            'gl': self.country,                 # country of the search
+            'hl': self.lang                     # language of the search
+        }
+
+        search = GoogleSearch(params)           # data extraction on the SerpApi backend
+        results = search.get_dict()             # JSON -> Python dict
+        
+        auto_complete_results = [result.get('value') for result in results.get('suggestions', [])]
+        
+        return auto_complete_results
+
+
+    def get_related_searches(self) -> list:
+        params = {
+            'api_key': self.api_key,            # https://serpapi.com/manage-api-key
+            'engine': 'google',                 # search engine
+            'q': self.query,                    # search query
+            'google_domain': self.domain,       # Google domain to use
+            'gl': self.country,                 # country of the search
+            'hl': self.lang                     # language of the search
+        }
+
+        search = GoogleSearch(params)           # data extraction on the SerpApi backend
+        results = search.get_dict()             # JSON -> Python dict
+        
+        related_searches_results = [result.get('query') for result in results.get('related_searches', [])]
+
+        return related_searches_results
+
+
+    def get_related_questions(self, depth_limit: int = 0) -> list:
+        params = {
+            'api_key': self.api_key,            # https://serpapi.com/manage-api-key
+            'engine': 'google',                 # search engine
+            'q': self.query,                    # search query
+            'google_domain': self.domain,       # Google domain to use
+            'gl': self.country,                 # country of the search
+            'hl': self.lang                     # language of the search
+        }
+
+        search = GoogleSearch(params)           # data extraction on the SerpApi backend
+        results = search.get_dict()             # JSON -> Python dict
+
+        related_questions_results = [result.get('question') for result in results.get('related_questions', [])]
+
+        if depth_limit > 4:
+            depth_limit = 4
+
+        if depth_limit:
+            def get_depth_results(token: str, depth: int) -> None:
+                ''' This function allows you to extract more data from People Also Ask.
+                
+                The function takes the following arguments:
+                
+                - `token` - allows access to additional related questions;
+                
+                - `depth` - limits the input depth for each related question. '''
```

Comment:
```suggestion
                :param token:  allows access to additional related questions;
                :param depth: limits the input depth for each related question. '''
```

![image](https://user-images.githubusercontent.com/78694043/228190165-c2f4e5a6-a6b5-458a-8773-10f13b5bada5.png)